api/controllers/build-log.js

```diff
@@ -9,74 +10,75 @@ function decodeb64(str) {
   return null;
 }
 
-module.exports = {
-  create: (req, res) => {
-    Promise.resolve(Number(req.params.build_id))
-    .then((id) => {
-      if (isNaN(id)) { throw 404; }
-      return Build.findByPk(id);
-    })
-    .then((build) => {
-      if (!build) {
-        throw 404;
-      } else if (build.token !== req.params.token) {
-        throw 403;
-      }
-
-      return BuildLog.create({
-        build: build.id,
-        output: decodeb64(req.body.output),
-        source: req.body.source,
-      });
-    })
-    .then(buildLog => buildLogSerializer.serialize(buildLog))
-    .then((buildLogJSON) => { res.json(buildLogJSON); })
-    .catch((err) => {
-      res.error(err);
+module.exports = wrapHandlers({
+  create: async (req, res) => {
+    const { build_id: buildId, token } = req.params;
+
+    const build = await Build.findOne({ where: { id: buildId, token } });
```

Comment:
A lot of the logic appears to be gone, here's the thought process.
1. Include the token in the query for the build so we just get a 404 back and don't have to do an explicit check. This follows a line of thought that says to just return a 404 instead of 403 for authorization failure to prevent any information from being communicated to anyone with bad intent.
2. Allow the type conversion to be handled in Sequelize and throw an error which is eventually handled in a common place (see the change to the `error` response handler) and results in a 404.